spec/rubocop/cop/rspec/aggregate_examples_spec.rb

```diff
@@ -0,0 +1,790 @@
+RSpec.describe RuboCop::Cop::RSpec::AggregateExamples, :config do
+  subject(:cop) { described_class.new(config) }
+
+  let(:cop_config) { {} }
+
+  shared_examples 'detects and autocorrects' do |offensive_source, good_source|
+    it 'does not detect an offense in good_source code' do
+      expect_no_offenses(good_source)
+    end
+
+    it 'detects an offense in offensive source code' do
+      expect_offense(offensive_source)
+    end
+
+    bad_code_without_message =
+      offensive_source.lines.delete_if { |line| line =~ /\A\s*\^/ }.join
+    include_examples 'autocorrect', bad_code_without_message, good_source
```

Comment:
These shared examples could be rewritten to use `expect_correction`, e.g.
```rb
shared_examples 'detects and autocorrects' do |offensive_source, good_source|
  it 'does not detect an offense in good_source code' do
    expect_no_offenses(good_source)
  end

  it 'detects an offense in offensive source code' do
    expect_offense(offensive_source)
    expect_correction(good_source)
  end
end
```
But it’s probably better to do in a separate follow-up pull request.

Reply:
Learned about those helpers from your comment. Got rid of those shared examples. The change made the spec slightly less compact (+55 lines), but I think more readable.